src/util/system.cpp

```diff
@@ -353,23 +353,19 @@ const std::set<std::string> ArgsManager::GetUnsuitableSectionOnlyArgs() const
     return unsuitables;
 }
 
-
-const std::set<std::string> ArgsManager::GetUnrecognizedSections() const
+const std::list<SectionInfo> ArgsManager::GetUnrecognizedSections() const
 {
     // Section names to be recognized in the config file.
     static const std::set<std::string> available_sections{
         CBaseChainParams::REGTEST,
         CBaseChainParams::TESTNET,
         CBaseChainParams::MAIN
     };
-    std::set<std::string> diff;
 
     LOCK(cs_args);
-    std::set_difference(
-        m_config_sections.begin(), m_config_sections.end(),
-        available_sections.begin(), available_sections.end(),
-        std::inserter(diff, diff.end()));
-    return diff;
+    auto u = m_config_sections;
```

Comment:
Use better variable name. Maybe also drop `auto`.

Reply:
@promag I cannot drop ```auto``` simply. I am happy if you explain a little more.

Reply:
I mean, could have explicit type.

Reply:
Thanks. I understand.
Addressed.